### PR TITLE
feat: preparations for gas estimation

### PR DESCRIPTION
**Pull request by Rqnsom**
This commit is a mix of many things:
 - address handling improvment (tests modified)
 - a first test for publishing bundle written
   - a new Move package introduced (taken from move-vm-backend/tests/assets)
 - raw_publish_module / raw_publish_module bundle functions prepared: these functions will be used by the RPC procedures

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -597,13 +584,58 @@ pub mod pallet {
...
+
+        /// Publish the module using the appropriate gas strategy.
+        pub fn raw_publish_module(
+            address: &T::AccountId,
+            bytecode: Vec<u8>,
+            gas: GasStrategy,
+        ) -> Result<VmResult, Error<T>> {
+            let storage = Self::move_vm_storage();
+            let vm = Mvm::new(storage).map_err(|_err| Error::<T>::PublishModuleFailed)?;
+            let address = Self::to_move_address(address).unwrap();
```
Comment (asmie) [inline src/lib.rs:609]: That can never fail?

Maybe that could be better:
```suggestion
            let address = Self::to_move_address(address)?;
```

File: src/lib.rs [hunk 2]
```diff
@@ -597,13 +584,58 @@ pub mod pallet {
...
+
+        /// Publish the bundle using the appropriate gas strategy.
+        pub fn raw_publish_bundle(
+            address: &T::AccountId,
+            bundle: Vec<u8>,
+            gas: GasStrategy,
+        ) -> Result<VmResult, Error<T>> {
+            let storage = Self::move_vm_storage();
+            let vm = Mvm::new(storage).map_err(|_err| Error::<T>::PublishBundleFailed)?;
+            let address = Self::to_move_address(address).unwrap();
```
Comment (asmie) [inline src/lib.rs:624]: The same as the above.

File: tests/mock.rs [hunk 3]
```diff
@@ -79,3 +80,24 @@ frame_support::construct_runtime!(
         MoveModule: pallet_move,
     }
 );
+
+// Common constants accross the tests.
+#[allow(dead_code)]
+pub const INFINITE_GAS: u64 = u64::MAX;
+pub const CAFE_ADDR: &str = "0xCAFE";
+pub const BOB_ADDR: &str = "5FHneW46xGXgs5mUiveU4sbTyGBzmstUspZC92UhjJM694ty";
+lazy_static::lazy_static! {
```
Comment (asmie) [inline tests/mock.rs:89]: Nice! :)

**Review by asmie — CHANGES_REQUESTED** [review]
Looking good!
Let's make sure if that `unwrap` calls can be removed.

File: src/lib.rs (hunk 1 above)
Comment (Rqnsom) [reply src/lib.rs]: Done

File: src/lib.rs (hunk 2 above)
Comment (Rqnsom) [reply src/lib.rs]: Done

**Review by asmie — APPROVED** [review]

**Pull request merged**